### ISSUE TITLE
support MIDI hardware connected through interfaces?

### ISSUE THREAD

**Issue by adcxyz** [ModalityTeam/Modality-toolkit#241]
Hi all, 
to what extent do we want to support MIDI hardware without USB (say a master keyboard or faderbox) 
which is connected through MIDI plugs on a MIDI interface? 
For that case, we would need desc files that describe just the MIDI device elements, but without an idInfo. On making that device, we would have to supply the idInfo from the interface's USB port, e.g. from a Fireface MIDIIn port that has the physical MIDI plug the thing is connected to.
Opinions?

**Comment by LFSaw** [comment]
I'd appreciate this.

**Comment by bhabbestad** [comment]
I second that.

b

10. okt. 2015 kl. 00:21 skrev LFSaw <notifications@github.com>:

**Comment by LFSaw** [comment]
I deleted my assignment and moved the milestone release to 0.9.5

**Comment by adcxyz** [comment]
Got a first pass at this working - from joybox desc file: 

Example for a MIDI hardware device and its elements.
	The joybox is made from a Doepfer pocket electronics kit,
	which can send 16 midi controls,
	and 8 XY joysticks in a 4x2 layout.

```
// usage example:
// make an Mktl for it - will say that this is virtual only
MKtl(\jbx, "decampo-joybox");
MKtl(\jbx).gui;
// MKtl has as idInfo : "EXTERNAL", which is not an existing port name.
MKtl(\jbx).desc.fullDesc.at(\idInfo);

// So we need to tell it which midi hardware interface we connect it to.
// See which MIDI sources are available:
MIDIClient.sources;
// figure out which one you want to use,
// and open it via the port name as idInfo:
MKtl(\jbx).openDeviceVia("IAC Driver");
```

**Comment by adcxyz** [comment]
closed with 7279ab0
